telemetry/src/ws.rs

```diff
@@ -561,35 +565,37 @@ mod tests {
             ..
         } = Suite::new();
 
-        // Fail sending the first message
-        fail_send.store(true, Ordering::Release);
-        telemetry_sender
-            .send(system_connected_telemetry())
-            .await
-            .unwrap();
-        assert!(message_receiver.try_next().is_err());
-        tokio::time::sleep(Duration::from_millis(100)).await;
-
-        // The second message is not sent because the sink is reset
-        fail_send.store(false, Ordering::Release);
-        telemetry_sender
-            .send(system_interval_telemetry(1))
-            .await
-            .unwrap();
-        assert!(message_receiver.try_next().is_err());
-        tokio::time::sleep(Duration::from_millis(100)).await;
-
-        // Fail sending the first message after reconnect
-        fail_send.store(true, Ordering::Release);
-        tokio::time::sleep(Duration::from_secs(1)).await;
-        assert!(message_receiver.try_next().is_err());
-
-        // The message is sent
-        fail_send.store(false, Ordering::Release);
-        tokio::time::sleep(Duration::from_secs(1)).await;
-        assert!(message_receiver.try_next().is_ok());
-
-        drop(telemetry_sender);
+        {
+            // Move `telemetry_sender` into inner scope
+            let telemetry_sender = telemetry_sender;
+            // Fail sending the first message
+            fail_send.store(true, Ordering::Release);
+            telemetry_sender
+                .send(system_connected_telemetry())
+                .await
+                .unwrap();
+            assert!(message_receiver.try_next().is_err());
+            tokio::time::sleep(Duration::from_millis(100)).await;
+
+            // The second message is not sent because the sink is reset
+            fail_send.store(false, Ordering::Release);
+            telemetry_sender
+                .send(system_interval_telemetry(1))
+                .await
+                .unwrap();
+            assert!(message_receiver.try_next().is_err());
+            tokio::time::sleep(Duration::from_millis(100)).await;
+
+            // Fail sending the first message after reconnect
+            fail_send.store(true, Ordering::Release);
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            assert!(message_receiver.try_next().is_err());
+
+            // The message is sent
+            fail_send.store(false, Ordering::Release);
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            assert!(message_receiver.try_next().is_ok());
+        }
```

Comment:
is this really better than previous solution?

Reply:
if you were to extract it into a function I wouldn't mind

Reply:
I will extract it into function.